private/pkg/storage/bucket.go

```diff
@@ -52,6 +52,38 @@ type ReadBucket interface {
 	Walk(ctx context.Context, prefix string, f func(ObjectInfo) error) error
 }
 
+// PutOptions are the possible options that can be passed to a Put operation.
+type PutOptions struct {
+	IfNotExists      bool
+	CustomChunkSize  bool
```

Comment:
Why is this needed over just checking if `ChunkSize` is not 0?

Reply:
Because `= 0` could have a meaning, like in GCS it is disable multi-part and upload it all in a single request.